### ISSUE TITLE
Gem fails to install on High Sierra

### ISSUE THREAD

**Issue by arikalish** [donmelton/video_transcoding#164]
I know Gramps don't beta (tm), but I figured since the release of macOS 10.13 is near I'd add this issue now.

When attempting to install the video_transcoding 0.17.4 on macOS 10.13 there is a type conversion error:

```
Media:~ media$ sudo gem install video_transcoding
Fetching: video_transcoding-0.17.4.gem (100%)
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String
```

Unfortunately, there's no line number given.

of note, 10.13's system Ruby is v2.3

```
Media:~ media$ gem -v
2.5.2
Media:~ media$ ruby --version
ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
Media:~ media$ 
```

**Comment by arikalish** [comment]
Some more details:

```
Media:~ media$ sudo gem install --backtrace video_transcoding
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:233:in `check_executable_overwrite'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:474:in `block in generate_bin'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:461:in `each'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:461:in `generate_bin'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:301:in `install'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/resolver/specification.rb:97:in `install'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:166:in `block in install'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:156:in `each'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:156:in `install'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:250:in `install_gem'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:300:in `block in install_gems'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `each'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `install_gems'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:203:in `execute'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/command.rb:308:in `invoke_with_build_args'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/command_manager.rb:169:in `process_args'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/command_manager.rb:139:in `run'
	/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/gem_runner.rb:55:in `run'
	/usr/bin/gem:21:in `<main>'
Media:~ media$ 

```

**Comment by arikalish** [comment]
And one more... this one points directly to the `.gemspec` file.

```
Media:~ media$ sudo gem install -V video_transcoding
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=video_transcoding
200 OK
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/LICENSE
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/README.md
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/bin/convert-video
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/bin/detect-crop
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/bin/query-handbrake-log
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/bin/transcode-video
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/cli.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/console.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/copyright.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/crop.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/errors.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/ffmpeg.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/handbrake.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/media.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/mkvpropedit.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/mp4track.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/tool.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/lib/video_transcoding/version.rb
/Library/Ruby/Gems/2.3.0/gems/video_transcoding-0.17.4/video_transcoding.gemspec
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String
```

Maybe the multiline string in the `description` field is throwing it off?

**Comment by arikalish** [comment]
OK... sorry to spam the issue. This seems to be a bug in RubyGems. Updating to 2.6.3 fixed the issue for me. I had to remove `/usr/local/bin/update_rubygems` per this comment: https://github.com/rubygems/rubygems/issues/1560#issuecomment-245198719

so:

```
Media:~ media$ rm /usr/local/bin/update_rubygems 
override rwxr-xr-x  root/admin for /usr/local/bin/update_rubygems? Y
Media:~ media$ sudo gem update -N --system 2.6.3
Updating rubygems-update
Successfully installed rubygems-update-2.6.3
Installing RubyGems 2.6.3
RubyGems 2.6.3 installed
.
.
.
Media:~ media$ sudo gem install video_transcoding
Successfully installed video_transcoding-0.17.4
Parsing documentation for video_transcoding-0.17.4
Installing ri documentation for video_transcoding-0.17.4
Done installing documentation for video_transcoding after 4 seconds
1 gem installed
```

Oy...

**Comment by donmelton** [comment]
@arikalish Wow! You filed and then resolved this issue while I was in the shower. :)

That's actually good to know because I'm sure you won't be the first person to hit that bug. Of course, this is exactly _why_ Gramps don't Beta. I'll probably get around to installing 10.13 in December or so. No rush!

**Comment by arikalish** [comment]
Yeah... curious is that RubyGems version is the system version or if that's something left over from my Sierra install. I'm assuming the former because it worked on 10.12.

**Comment by donmelton** [comment]
@arikalish You should always reinstall Homebrew when you install a new major OS. And install Ruby via that Homebrew after that. Then you shouldn't have a problem.

**Comment by arikalish** [comment]
Yeah... I did a ‘brew update’ and ‘brew upgrade’. I went down an ‘rvm’ rabbit hole a while back and never want to deal with that again ;).

**Comment by donmelton** [comment]
@arikalish Yeah, best to stay away from RVM. :)

**Comment by kvanh** [comment]
Just FYI, I did an install on a 10.13 GM install and the install completed successfully. I did need to install GM Xcode and Xcode command line tools for Homebrew to work correctly, but that seems unrelated to the gem install. This machine was originally built as a format and start from scratch install of High Sierra beta (I think beta 5?). GM install was on top of the beta builds.

Versions of gem and ruby reported are identical to first message in thread.

Dunno why anyone would be reluctant to install a beta Xcode on top of a beta OS :)
